AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -108,17 +108,19 @@ - (void)_layoutNodes:(NSArray *)nodes atIndexPaths:(NSArray *)indexPaths withAni
   for (NSUInteger j = 0; j < nodes.count && j < indexPaths.count; j += kASDataControllerSizingCountPerProcessor) {
     NSArray *subIndexPaths = [indexPaths subarrayWithRange:NSMakeRange(j, MIN(kASDataControllerSizingCountPerProcessor, indexPaths.count - j))];
     
-    // TODO: The current implementation does not make use of different constrained sizes per node.
     // There should be a fast-path that avoids all of this object creation.
     NSMutableArray *nodeBoundSizes = [[NSMutableArray alloc] initWithCapacity:kASDataControllerSizingCountPerProcessor];
     [subIndexPaths enumerateObjectsUsingBlock:^(NSIndexPath *indexPath, NSUInteger idx, BOOL *stop) {
-      [nodeBoundSizes addObject:[NSValue valueWithCGSize:[_dataSource dataController:self constrainedSizeForNodeAtIndexPath:indexPath]]];
+      ASSizeRange constrainedSize = [_dataSource dataController:self constrainedSizeForNodeAtIndexPath:indexPath];
```

Comment:
We should definitely optimize this in the near future, for the common case when there is not a different constrainedSize per node.

In most cases, developers will want to use e.g. the width of the ASTableView + CGFloatMax as their constrainedSize coming into measure: of their ASCellNode.  Boxing NSValues and adding to an array, plus all the method calls, is pretty inefficient.